app/parsers/bulkrax/csv_parser.rb

```diff
@@ -209,11 +210,30 @@ def key_allowed(key)
 
     # All possible column names
     def export_headers
-      headers = ['id']
-      headers << source_identifier.to_s
-      headers << 'model'
-      headers << 'collections'
-      importerexporter.mapping.each_key { |key| headers << key if key_allowed(key) }
+      headers = []
+      # the object keys are not part of a work types default metadata but the object itself is,
+      # so we check according to it and add the acceptable headers
+      importerexporter.mapping.each do |key, value|
+        if value.key?('object') && key_allowed(value['object'])
+          object_keys = self.headers.each { |sh| sh.to_s.include?(value['object']) }
```

Comment:
this code will (unless it throws an exception) always do the same thing as `object_keys = self.headers`.  The loop here appears to be just for fun.

Reply:
`each` returns every item in the iterator so `[1, 2, 3, 4].each {|i| i if i.odd?}` returns `[1,2,3,4]`.  `[1,2,3,4].detect {|i| i.odd?}` would return `[1,3]` `[1,2,3,4].map {|i| i if i.odd?}` would return `[1,nil,3,nil]`.  I think its the detect that you wanted, but then that code got changed a lot in #339

Reply:
gotcha. I started off with `each` and then changed the method to `select` on friday for the same reason you stated. but something started messing up with the headers. I don't remember what. so I changed it back.